browser/permissions/google_sign_in_permission_browsertest.cc

```diff
@@ -0,0 +1,556 @@
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), embedding_url_));
+  content::TestNavigationObserver error_observer(contents());
+
+  ClickButtonWithId(kAuthButtonHtmlId);
+  WaitForLoadStopWithoutSuccessCheck(contents());
+  error_observer.Wait();
+  // Navigation blocked
+  EXPECT_EQ(false, error_observer.last_navigation_succeeded());
+  EXPECT_EQ(net::ERR_BLOCKED_BY_CLIENT, error_observer.last_net_error_code());
+  // No additional prompt shown
+  EXPECT_EQ(1, prompt_factory()->show_count());
+}
+
+IN_PROC_BROWSER_TEST_F(GoogleSignInBrowserTest, PermissionDismiss) {
+  SetGoogleSignInPref(true);
+  EXPECT_EQ(0, prompt_factory()->show_count());
+
+  // Try to set 3p cookies from auth domain, should not work
+  NavigateToPageWithFrame(https_cookie_iframe_url_);
+  ExpectCookiesOnHost(GURL(kAccountsGoogleUrl), "");
+  NavigateFrameTo(google_oauth_cookie_url_);
+  ExpectCookiesOnHost(GURL(kAccountsGoogleUrl), "");
+
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), embedding_url_));
+
+  EXPECT_EQ(content_settings()->GetContentSetting(
+                GURL(kAccountsGoogleUrl), embedding_url_,
+                ContentSettingsType::BRAVE_GOOGLE_SIGN_IN),
+            ContentSetting::CONTENT_SETTING_ASK);
+  EXPECT_EQ(cookie_settings()->GetCookieSetting(
+                GURL(kAccountsGoogleUrl), embedding_url_, nullptr,
+                content_settings::CookieSettingsBase::QueryReason::kCookies),
+            ContentSetting::CONTENT_SETTING_BLOCK);
+
+  prompt_factory()->set_response_type(
+      permissions::PermissionRequestManager::DISMISS);
+
+  // Have website issue request for Google auth URL
+  ClickButtonWithId(kAuthButtonHtmlId);
+  prompt_factory()->WaitForPermissionBubble();
+
+  // Confirm that content settings did not change
+  EXPECT_EQ(content_settings()->GetContentSetting(
+                embedding_url_, embedding_url_,
+                ContentSettingsType::BRAVE_GOOGLE_SIGN_IN),
+            ContentSetting::CONTENT_SETTING_ASK);
+  EXPECT_EQ(cookie_settings()->GetCookieSetting(
+                GURL(kAccountsGoogleUrl), embedding_url_, nullptr,
+                content_settings::CookieSettingsBase::QueryReason::kCookies),
+            ContentSetting::CONTENT_SETTING_BLOCK);
+
+  EXPECT_EQ(1, prompt_factory()->show_count());
+}
+
+IN_PROC_BROWSER_TEST_F(GoogleSignInBrowserTest, PrefTurnedOff) {
+  SetGoogleSignInPref(false);
+  EXPECT_EQ(0, prompt_factory()->show_count());
+
+  // Try to set 3p cookies from auth domain, should not work
+  NavigateToPageWithFrame(https_cookie_iframe_url_);
+  ExpectCookiesOnHost(GURL(kAccountsGoogleUrl), "");
+  NavigateFrameTo(google_oauth_cookie_url_);
+  ExpectCookiesOnHost(GURL(kAccountsGoogleUrl), "");
+
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), embedding_url_));
+  content::TestNavigationObserver error_observer(contents());
+
+  ClickButtonWithId(kAuthButtonHtmlId);
+  WaitForLoadStopWithoutSuccessCheck(contents());
+  error_observer.Wait();
+  // Navigation blocked
```

Comment:
this doesn't look good. Is it really a good idea to block some web pages when some feature is disabled? Did we block the page with disabled "Google Sign-In" toggle before this new permission feature? Why not just open the auth page with the usual 3p-ephemeral cookies?

Reply:
hm yeah, this was an explicit decision made early-on, but I can't recall why and doesn't seem like a good idea. I'll fix

Reply:
Fixed, let me know what you think